ui/src/pages/AllStories/reducer.ts

```diff
@@ -122,5 +176,114 @@ export function allStoriesReducer(state: State, action: Action): State {
       };
     }
+    case "SET_CHANGED_VISIBILITY": {
+      return {
+        ...state,
+        changedVisibility: action.data,
+      };
+    }
+    case "SET_VISIBLE_TABLE_STATE": {
+      return {
+        ...state,
+        visibleTableState: action.data,
+      };
+    }
+    case "HANDLE_SEARCH/FILTER": {
+
+      const newState = action.data
+
+      const requestSearchHelper = (row) => {
+        if (newState.search == "") {
+          return true
+        }
+        let doesExist = false;
+        Object.keys(row).forEach((prop) => {
+          //Exclude search for StoryView members not displayed on table cells
+          const excludedParameters =
+            prop !== "image_url" &&
+            prop !== "video_url" &&
+            prop !== "content" &&
+            prop !== "is_visible";
+          const numExist =
+            typeof row[prop] === "number" &&
+            row[prop].toString().includes(newState.search) &&
+            excludedParameters;
+          const stringExist =
+            typeof row[prop] === "string" &&
+            row[prop].toLowerCase().includes(newState.search.toLowerCase()) &&
+            excludedParameters;
+          if (stringExist || numExist) {
+            doesExist = true;
+          }
+        });
+        return doesExist;
+      };
+
+      const handleTagFilterHelper = (row) => {
+        let exist = false
+        let zeroChecked = true //atleast one tag should be checked off
+        Object.keys(newState.filterState.tags).forEach((tag) => {
+          if (newState.filterState.tags[tag] ) {
+            zeroChecked = false
+            if ( row.tags.includes(tag.toString())) {
+              exist = true;
+            }
+          } 
+        });
+        if (zeroChecked) {
+          return true
+        }
+        return exist;
+      };
+    
+      const checkRowVisibility = (row) => {
+        console.log(newState.filterState.visibility)
+        return ((newState.filterState.visibility.visible && row.is_visible) || (newState.filterState.visibility.nonVisible && !row.is_visible) || (!newState.filterState.visibility.nonVisible && !newState.filterState.visibility.visible))   
+      }
+    
+      const requestSearchAndFilter = () => {
+        if (state.tabValue === 0) {
+          let filteredRows: StoryView[] = newState.origTableData.filter((row) => {
+            return requestSearchHelper(row);
+          });
+          filteredRows = filteredRows.filter((row) => {
+            return checkRowVisibility(row)
+          });
+          filteredRows = filteredRows.filter((row) => {
+            return handleTagFilterHelper(row)
+          });
+          newState.tableData = filteredRows
+        } else if (state.tabValue === 1) {
+          let filteredRows: StoryView[] = newState.visibleTableFilterState.filter(
+            (row) => {
+              return requestSearchHelper(row);
+            }
+          );
+          filteredRows = filteredRows.filter((row) => {
```

Comment:
Should `filteredRows = filteredRows.filter((row) => {
            return checkRowVisibility(row)
          });
          filteredRows = filteredRows.filter((row) => {
            return handleTagFilterHelper(row)
          });`

be extracted into its own helper function? A lot of repeated code here